bin/worker.js

```diff
@@ -0,0 +1,16 @@
+import { reportMetric } from "../lib/influx.js";
+
+export const handleRequest = async (request, env) => {
+  const url = new URL(request.url);
+  const newRequest = new Request(`${env.REQUEST_URL}${url.pathname}${url.search}`, request);
+
+  return await fetch(newRequest);
+}
```

Comment:
I am struggling to understand why we have to rewrite the request URL.

In the examples I am seeing in CF docs, they always call `fetch(request)` with the original URL.

What am I missing?

See e.g. here: https://developers.cloudflare.com/workers/runtime-apis/context/#waituntil


```js
export default {
  async fetch(request, env, ctx) {
    // Forward / proxy original request
    let res = await fetch(request);

    // Add custom header(s)
    res = new Response(res.body, res);
    res.headers.set('x-foo', 'bar');

    // Cache the response
    // NOTE: Does NOT block / wait
    ctx.waitUntil(caches.default.put(request, res.clone()));

    // Done
    return res;
  },
};
```

Reply:
https://developers.cloudflare.com/workers/reference/how-the-cache-works/#fetch

> In the context of Workers, a [fetch](https://developers.cloudflare.com/workers/runtime-apis/fetch/) provided by the runtime communicates with the Cloudflare cache. First, fetch checks to see if the URL matches a different zone. If it does, it reads through that zone’s cache (or Worker). Otherwise, it reads through its own zone’s cache, even if the URL is for a non-Cloudflare site. Cache settings on fetch automatically apply caching rules based on your Cloudflare settings. fetch does not allow you to modify or inspect objects before they reach the cache, but does allow you to modify how it will cache.
When a response fills the cache, the response header contains CF-Cache-Status: HIT. You can tell an object is attempting to cache if one sees the CF-Cache-Status at all.
This [template](https://developers.cloudflare.com/workers/examples/cache-using-fetch/) shows ways to customize Cloudflare cache behavior on a given request using fetch.

Reply:
This was confusing me, but I managed to wrap my head around it after reading [this](https://developers.cloudflare.com/workers/configuration/routing/routes/). I had understanding that worker would act more as a reverse proxy so forwarding request would make the worker call themself again and again. 

Well, apparently it doesn't work this way. If you set your cloudflare worker to a route already proxied by cloudflare it will still be able to forward the request.

Reply:
> This was confusing me, but I managed to wrap my head around it after reading [this](https://developers.cloudflare.com/workers/configuration/routing/routes/). 

Great find! 

> Routes are a set of rules that evaluate against a request's URL. Routes are recommended for you if you have a designated application server you always need to communicate with. Calling fetch() on the incoming Request object will trigger a subrequest to your application server, as defined in the DNS settings of your Cloudflare zone.

That's what I was expecting/looking for 👍🏻